tool/lib/commands/update_flutter_sdk.dart

```diff
@@ -33,22 +32,14 @@ const _useCacheFlag = 'use-cache';
 /// `devtools_tool update-flutter-sdk [--from-path] [--no-use-cache]`
 class UpdateFlutterSdkCommand extends Command {
   UpdateFlutterSdkCommand() {
-    argParser
-      ..addFlag(
-        _fromPathFlag,
-        abbr: 'p',
-        negatable: false,
-        help: 'Update the Flutter SDK that is on PATH (your local '
-            'flutter/flutter git checkout)',
-      )
```

Comment:
I think this should still be optional. We may not always want to wipe our local flutter checkout. Maybe we should change it to `--on-path` too.

Reply:
> Uses the Flutter SDK from PATH to run the Processes started by the update-flutter-sdk command.
> vs
> Uses the `tool/flutter-sdk` to run the Processes started by the update-flutter-sdk command.

I guess my confusion is that it's not clear why we need this distinction. The two things you list that those commands do appears the same (update both Flutter SDKs). What's the purpose of running with the local tool/flutter-sdk SDK when updating the one from PATH?

I was trying to keep things simple by saying that basically you're always using one SDK or the other... Only one SDK exists in the world of the script, and you can choose the one from PATH or the one from the current running VM (which is essentially always tool/flutter-sdk because the shell scripts force that). Mixing them feels like it could cause confusion, because you need to know/remember which SDK is being used for what.


> Another way to summarize all this, if the intent of --flutter-from-path for all devtools_tool commands means "use my Flutter SDK on PATH for all processes and subprocesses started by this command", it feels strange for this flag to also mean "but when you run with this particular command update-flutter-sdk it actually does other things too".

I agree, and my original goal was for it not to do anything different at all, but rather just update whichever SDK was being used (and updated). However the original implementation seemed to update _both_ SDKs and I wasn't sure if it reasonable to change that. In summary:

- `--flutter-from-path` means always use the Flutter SDK from path for all spawned processes - Flutter or Dart, including devtools_tool. This occurs all the way down.
- `update-flutter-sdk` will always update the "current" SDK (based on that flag above), but additionally it will ensure "tool/flutter-sdk" is always up-to-date if it wasn't the one already being updated.

My preference would've been to remove that last part, (so update-flutter-sdk only ever updates one SDK - the one being run) because I think it's much easier to understand. If you want to update both SDKs, you could run `devtools_tool update-flutter-sdk && devtools_tool update-flutter-sdk --flutter-from-path`). I'm not sure if that addresses everything you wanted though?